### PR TITLE
Week numbers

### LINKED ISSUE #41: Week numbers
Would be pretty neat to have a switch add week numbers. For example:

```sh
ᐅ ncal -w
    October 2022
Mo     3 10 17 24 31
Tu     4 11 18 25
We     5 12 19 26
Th     6 13 20 27
Fr     7 14 21 28
Sa  1  8 15 22 29
Su  2  9 16 23 30
   39 40 41 42 43 44
```

### PR DESCRIPTION
**Pull request by elliotwutingfeng**
Implements/closes #41.

image

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -211,7 +228,36 @@ pub fn calendar(year: u32, locale_str: &str, starting_day: u32) -> Vec<Vec<Vec<S
...
+                if line < 2 {
+                    rows[row_counter][column_counter][line] =
+                        "   ".to_string() + &rows[row_counter][column_counter][line]
+                } else if !&rows[row_counter][column_counter][line].trim().is_empty() {
+                    let padding = if week_counter > 9 {
+                        " ".to_string()
+                    } else {
+                        "  ".to_string()
+                    };
+                    rows[row_counter][column_counter][line] = padding
```
Comment (arthurhenrique) [inline src/lib.rs:245]: I think this could be write using `format!` macro, its more eficient

File: src/lib.rs [hunk 2]
```diff
@@ -211,7 +228,36 @@ pub fn calendar(year: u32, locale_str: &str, starting_day: u32) -> Vec<Vec<Vec<S
             starting_day,
             locale_info.month_names(),
             locale_info.week_day_names(),
+            week_numbers,
         );
+
+        if week_numbers {
+            (0..rows[row_counter][column_counter].len()).for_each(|line| {
```
Comment (arthurhenrique) [inline src/lib.rs:235]: I'll refactor further it to casual for loop, its more idiomatic.

**Review by arthurhenrique — APPROVED** [review]
Thanks for your work! ❤️

**Comment by elliotwutingfeng** [comment]
Just realised I introduced a bug; the current date is off by 1. Looking into it now...

**Pull request merged**